testing/python/fixtures.py

```diff
@@ -4534,8 +4534,178 @@ def test_fixt(custom):
     assert result.ret == ExitCode.TESTS_FAILED
 
 
+def test_fixture_info_after_dynamic_parametrize(pytester: Pytester) -> None:
```

Comment:
Are you sure that `test_fixture_info_after_dynamic_parametrize` passes? The rest are to make sure the correct behavior isn't going to change but this one should fail, because the `prune_dependency_tree` takes place after `fixtures.py::pytest_generate_tests`'s parametrization on `fixture3` and its two generated items along with the two from dynamic parametrization on `fixture2` make up four items causing the test to fail.
I added the test to main branch and it failed expectedly.
OK, I add docstring to the tests.

Reply:
Update: `test_reordering_after_dynamic_parametrize` had a bug. I fix it.